### ISSUE TITLE
Are WithName, WithValues, WithCallDepth thread safe methods?

### ISSUE THREAD

**Issue by pachanga** [go-logr/logr#154]
Hi! 

I'm looking at the documentation and it's not clear to me if these methods are safe to be used concurrently. I browsed sources of zapr implementation and it seems like it's not thread safe... Could you please elaborate a bit more on proper concurrent usage of the library? Thanks!

**Comment by wojas** [comment]
The Logger is passed as a value everywhere and not mutated after creation. If the LogSink implementation is safe for concurrent usage, the `logr.Logger` will be too. If you find this is untrue in some part of the Logger, this is a bug.

If there are parts of the zapr LogSink that you believe are not thread-safe, please do file an issue in the zapr repository.

**Comment by pachanga** [comment]
Yep, thanks for the reply, I'll file an issue in the zapr repository
